File: src/prompt/authorsNote.ts
import type { Message } from "./template";

/**
 * 将 Author's Note 以 system 消息形式注入到对话历史中
 *
 * @param history 当前对话历史
 * @param note Author's Note 文本
 * @param depth 注入深度：0 = 末尾，N = 倒数第 N 条消息之前
 */
export function injectAuthorNote(
  history: Message[],
  note: string,
  depth: number
): Message[] {
  if (!note.trim()) return history;

  const result = [...history];
  const insertIdx = Math.max(0, result.length - depth);
  result.splice(insertIdx, 0, {
    role: "system" as const,
    content: `[Author's Note]\n${note}`,
  });
  return result;
}

File: fake/strange/backend/app/prompt/authors_note.py
"""Author's Note 注入。"""
from typing import List


def inject_author_note(history: List[dict], note: str, depth: int) -> List[dict]:
    if not note.strip():
        return history

    result = list(history)
    insert_idx = max(0, len(result) - depth)
    result.insert(insert_idx, {"role": "system", "content": f"[Author's Note]\n{note}"})
    return result
